vkcs/internal/services/networking/v2/routers/requests.go

```diff
@@ -18,7 +25,47 @@ func (opts RouterCreateOpts) ToRouterCreateMap() (map[string]interface{}, error)
 	return util.BuildRequest(opts, "router")
 }
 
+const creationRetriesNum = 3
+const creationRetryDelay = 3 * time.Second
+
+func RetryNetworkingRouterCreation(_ context.Context, _ string, _ string, _ *gophercloud.RequestOpts, err error, failCount uint) error {
+	if failCount >= creationRetriesNum {
+		return err
+	}
+
+	if needRetryOnNetworkingRouterCreationError(err) {
+		return nil
+	}
+
+	return err
+}
+
+func needRetryOnNetworkingRouterCreationError(err error) bool {
+	if errutil.IsNotFound(err) {
+		return true
+	}
+
+	var http409Err gophercloud.ErrDefault409
+
+	if errors.As(err, &http409Err) {
+		neutronError, decodeErr := inetworking.DecodeNeutronError(http409Err.ErrUnexpectedResponseCode.Body)
+		if decodeErr != nil {
+			log.Printf("[DEBUG] failed to decode a neutron error: %s", decodeErr)
+			return false
+		}
+		if neutronError.Type == "NeutronDbObjectDuplicateEntry" {
+			time.Sleep(creationRetryDelay)
+			return true
+		}
+
+		return false
+	}
+
+	return false
+}
+
 func Create(c *gophercloud.ServiceClient, opts routers.CreateOptsBuilder) routers.CreateResult {
+	c.RetryFunc = RetryNetworkingRouterCreation
```

Comment:
You should not overwrite client's retry function here, as this at least vanishes default behavior of retrying 5xx errors.

Reply:
add new func CreateWithRetry